File: infosupport/src/app/models/appointment.ts
import {Time} from "@angular/common";

export class Appointment {

  appointment_code: number;
  start_time: Date;
  end_time: Date;
  is_digital: boolean;
  description: string;
  location: string;
  is_follow_up: boolean;
  big_code: number;
  patient_user_id: number;
  title: string
  constructor( start_time?: Date, end_time?: Date, is_digital?: boolean, description?: string, location?: string, is_follow_up?: boolean, big_code?: number, patient_user_id?: number, appointment_code?: number, title?: string) {
    this.start_time = start_time;
    this.end_time = end_time;
    this.is_digital = is_digital;
    this.description = description;
    this.location = location;
    this.is_follow_up = is_follow_up;
    this.big_code = big_code;
    this.patient_user_id = patient_user_id;
    this.appointment_code = appointment_code;
    this.title = title
  }

//Possible methods go here
}

File: infosupport-BE/src/main/java/nl/hva/models/Appointment.java
package nl.hva.models;

import com.fasterxml.jackson.annotation.JsonFormat;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "Appointment")
public class Appointment {

  @Id
  @GeneratedValue(strategy=GenerationType.IDENTITY)
  private Integer appointment_code;

  private LocalDateTime start_time;
  private LocalDateTime end_time;

  private Boolean is_digital;
  private String description;
  private String location;
  private Boolean is_follow_up;
  private Integer big_code;
  private String title;
  private Integer patient_user_id;



  public LocalDateTime getStart_time() {
    return start_time;
  }
  public LocalDateTime getEnd_time() {
    return end_time;
  }

  public boolean isIs_digital() {
    return is_digital;
  }

  public String getDescription() {
    return description;
  }

  public String getLocation() {
    return location;
  }

  public Boolean getIs_follow_up() {
    return is_follow_up;
  }

  public Integer getBig_code() {
    return big_code;
  }

  public Integer getPatient_user_id() {
    return patient_user_id;
  }

  public String getTitle() {
    return title;
  }

  public Integer getAppointment_code() {
    return appointment_code;
  }


}
